### ISSUE TITLE
Tracking an inherited model (abstract class)

### ISSUE THREAD

**Issue by diogolr** [treyhunner/django-simple-history#269]
As seen in https://github.com/treyhunner/django-simple-history/pull/112 and [docs](http://django-simple-history.readthedocs.io/en/latest/advanced.html#allow-tracking-to-be-inherited) the ``HistoricalRecords`` can now be used with ``inherited = True`` to tracking inherited models "automatically".

Despite this, I came across an issue with ``Django 1.10.6`` and the updated version of this repository ``master`` branch (https://github.com/treyhunner/django-simple-history/commit/d76e446c3c45dbc81a9ac6db9297c5649e27c1b4).

The issue can be reproduced with this example:

```
django-admin startproject myproject
cd myproject
django-admin startapp myapp
touch myproject/models.py
```

*myproject/models.py*
```python
# -*- coding: utf-8 -*-
from django.db import models
from simple_history.models import HistoricalRecords


class GenericModel( models.Model ):
    class Meta:
        abstract = True

    history = HistoricalRecords( inherit = True )
```

*myapp/models.py*
```python
# -*- coding: utf-8 -*-
from myproject.models import GenericModel


class MyModel( GenericModel ):
    class Meta:
        app_label = 'myapp'
```

*myproject/settings.py*
```python
...
INSTALLED_APPS = [
    ...
    'simple_history',
    'myapp',
]
...
MIDDLEWARE = [
    ...
    'simple_history.middleware.HistoryRequestMiddleware',
]
```

After that, run ``python manage.py makemigrations`` and the following error will be raised:
```
RuntimeError: Model class myproject.models.HistoricalMyModel doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
```

Any ideas about what is going on?

**Comment by rocktavious** [comment]
Bump - I am experiencing this too and would like to know how to solve - it seems related to the code in https://github.com/treyhunner/django-simple-history/blob/master/simple_history/models.py function create_history_model

**Comment by tolomea** [comment]
This seems to work:

    patchy.patch(simple_history.models.HistoricalRecords.get_meta_options, """
    @@ -13,4 +13,5 @@
             name = string_concat('historical ',
                                  smart_text(model._meta.verbose_name))
         meta_fields['verbose_name'] = name
    +    meta_fields['app_label'] = model._meta.app_label
         return meta_fields
    """)

not sure if it's correct, I don't really understand what create_history is trying to do with app_lablels.

**Comment by diogolr** [comment]
@tolomea, I think that's because the historical tables should be created at the same app (like a new model).

This fix is already commited/accepted? What's the commit number/reference?

**Comment by tolomea** [comment]
@diogolr 

I didn't quite follow you there, what I've posted is a monkey patch to simple history to force the history table to have the same app label as the model it's tracking the history of.

**Comment by adamchainz** [comment]
A [patchy](https://github.com/adamchainz/patchy) patch is one better than a monkey patch 👌

**Comment by rossmechanic** [comment]
@tolomea that fails a few of the tests in the django-simple-history test suite. Essentially what you're doing there is overriding the given app_label. If `HistoricalRecords` is specified as a field in your models, then this is fine. But if you use the `register` function with a custom `app` parameter, then this will overwrite that `app` parameter. Adding a PR that provides the change conditionally, based on whether the model used `HistoricalRecords` as a field or whether the `register` function was used

**Comment by rossmechanic** [comment]
This was fixed in #355
